docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,66 @@
+# Release Versioning / Branching Strategy
+
+## Context
+With `dbt` ever evolving and progressing, we continue to ship and release versions on a regular basis. As we release `dbt`, versioning and branching become key to identifying what changes for users and conveying the expectations of those changes. We currently push all our changes to `main` which is the latest version of our code. We don't always want to release the latest version of our code though. We want to be able to isolate changes to go into particular releases based on urgency (e.g. bug fixes) and stability (e.g. new features). 
+
+We already have an established versioning strategy when it comes to major/minor/patch releases. The following is our approach as to what to expect from each release version:
+
+* Major: This signifies that breaking changes which require user action are included in this release.
+* Minor version: This signifies that new features are included in this release.
+* Patch version: This signifies that bug and security fixes are included in this release.
+
+We need a versioning strategy that allows us to to convey to users the confidence and stability of code they are installing. We want a versioning strategy that allows users to try out new changes, and experiment with them to get early feedback. We want to make sure users can also get a preview of new features we are working on, so they can incorporate these changes into their work as soon as possible. 
+
+We need a branching strategy to support the goals of the versioning strategy. We do not want to have to stop development on `main` as we await a release, nor do we want every change in `main` published in every release. Using branches, we can decide which changes go into which release.
+
+## Requirements
+The following are requirements that must be met for releasing dbt:
```

Comment:
I think we need to make further distinctions here for the target of the release (pypi, docker, github, etc) because not all of them fit into this list (docker images specifically)

Reply:
Edit:  Is target what you meant by `type`?

Reply:
@iknox-fa I'm changing type to phase. I used the wrong term.

Also can you elaborate on what you would like to see in requirements for the release targets? Do you mean that Docker always releases from the `main` branch? I was thinking of creating a new doc specifically for the release targets but do you think that will impact the statements made in the requirements here?

Reply:
I added another requirement bullet point to define this. @jtcohen6 would you agree with my last point of what "released" means to us?